src/server/game/Entities/Player/Player.cpp

```diff
@@ -7534,6 +7534,23 @@ void Player::SendLoot(ObjectGuid guid, LootType loot_type)
     PermissionTypes permission = ALL_PERMISSION;
 
     LOG_DEBUG("loot", "Player::SendLoot");
+
+    std::array<AuraType, 3> toRemove = { SPELL_AURA_MOD_INVISIBILITY, SPELL_AURA_FEIGN_DEATH };
+    if (loot_type != LOOT_PICKPOCKETING)
+    {
+        toRemove = { SPELL_AURA_MOD_INVISIBILITY, SPELL_AURA_FEIGN_DEATH, SPELL_AURA_MOD_STEALTH };
```

Comment:
The logic here looks more complex than it needs to be, why don't we put the FOR inside this IF? We don't really need to filter specific aura types, you can't pickpocket while invisible or in feign death f.e

Reply:
Hm, good idea